### PR TITLE
Editors: Refactor Jetpack Redirect for Launchpad modal

### LINKED ISSUE #72615: Blocks: Post Editor Launchpad on save modal Follow up
#### Summary
As part of working on the [Blocks: Show the Launchpad on save modal inside the Site Editor](https://github.com/Automattic/jetpack/pull/28555) for all flows we have manually added the mapping entries for each flow inside https://mc.a8c.com/jetpack-crew/redirects/.

But it looks like we could set this up in a way where we don't need to re-register server side for each new intent we want to enable (see the `calypso-edit-post` example shown at the bottom of https://github.com/Automattic/jetpack/tree/trunk/projects/js-packages/components/tools/jp-redirect)

### PR DESCRIPTION
**Pull request by sharpirate**
Fixes https://github.com/Automattic/wp-calypso/issues/72615

## Proposed changes:
Refactor the Jetpack Redirect to use dynamic `[path]` so we only need a single redirect.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:
• Run `bin/jetpack-downloader test jetpack add/launchpad_modal_redirect` to sync this branch to your sandbox
• Sandbox `public-api.wordpress.com` and a Launchpad-enabled site
• Make a change in the Site or Post editor
• Once the Launchpad modal appears, click the "Next Steps" button
• Confirm you get redirected to Launchpad

### CODE REVIEW

**Review by agrullon95 — COMMENTED** [review]
I applied the changes to my sandbox, published/updated a post on my launchpad enabled site, clicked on the `Next Steps` button in the launchpad modal, and was redirected to the launchpad screen. Is there any way to confirm that the changes have actually applied correctly? This is the redirection URL when I click the Next steps button:

https://jetpack.com/redirect/?source=wpcom-launchpad-setup-build&query=siteSlug%3D{siteSlug}

**Comment by sharpirate** [comment]
@agrullon95 
I think the changes were not applied because you should see the generic `wpcom-launchpad-setup` in the URL and not the specific flow name `wpcom-launchpad-setup-build` as in your case.

This is what I see:
`https://jetpack.com/redirect/?source=wpcom-launchpad-setup&path=link-in-bio&query=siteSlug%3Dliblib9.wordpress.com`

**Comment by agrullon95** [comment]
@Sharpirate - I retested and now im seeing the correct URL: https://jetpack.com/redirect/?source=wpcom-launchpad-setup&path=link-in-bio&query=siteSlug%3Dfakesite.wordpress.com. Thanks.

**Review by agrullon95 — DISMISSED** [review]

**Review by jeherve — DISMISSED** [review]

**Comment by edanzer** [comment]
This is testing as expected. I tested changes in the site and post editor. In both cases, the Nexts Steps button/link points to the redirect with `wpcom-launchpad-setup` and clicking the button successfully navigates back to Launchpad. 

I'm approving this, but the checks below show this will need a merge/rebase and likely another approval to merge.

**Review by edanzer — DISMISSED** [review]

**Comment by edanzer** [comment]
Just retested this again after merge - still good.

**Review by edanzer — APPROVED** [review]

**Pull request merged**
